### PR TITLE
PP-1096 daemon protect (Linux OOM protection function) is wrong

### PR DESCRIPTION
**Pull request by vstumpf**
#### Issue-ID

* **[PP-1096](https://pbspro.atlassian.net/browse/PP-1096)**

#### Problem description
* daemon_protect writes to `/proc/<PID>/oom_score_adj`, and when it does, it writes the entire content of the buffer instead of just the length of the string. This showed up when compiling with AddressSanitizer.

#### Cause / Analysis
* It uses sizeof() instead of strlen(). 

#### Solution description
* Change sizeof()-1 to strlen().

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [x] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: src/lib/Libutil/daemon_protect.c [hunk 1]
```diff
@@ -62,15 +63,15 @@ daemon_protect(pid_t pid, enum PBS_Daemon_Protect action)
 	 */
 	snprintf(fname, MAXPATHLEN, oom_protect_new.oom_path, pid);
 	if ((fd = open(fname, O_WRONLY)) != -1) {
-		write(fd, oom_protect_new.oom_value[(int)action], sizeof(oom_protect_new.oom_value[(int)action])-1);
+		write(fd, oom_protect_new.oom_value[(int)action], strlen(oom_protect_new.oom_value[(int)action]));
```
Comment (mike0042) [inline src/lib/Libutil/daemon_protect.c]: In addition, I think we want to open the file as O_WRONLY|O_TRUNC
Same below as well.

**Comment by vstumpf** [comment]
Thanks @mike0042!

**Comment by vstumpf** [comment]
Thanks @prakashcv13!

@mike0042, I've squashed and rebased. Can you take this forward please?

**Comment by vstumpf** [comment]
Thanks!

**Pull request merged**